### PR TITLE
[Java] DataInput implementation over DirectBuffer

### PR DESCRIPTION
**Pull request by tkowalcz**
This PR addresses #273. I have added `DirectBufferDataInput` which implements JDK interface `DataInput`. This implementation adheres to the `DataInput` specification which, sadly, includes throwing checked EOF exception on underflow and being big endian by default. User can change byte order to make it read Agrona buffer's generated data. 

As a bonus it has few additional methods to read Agrona style strings into an `Appendable`.

### CODE REVIEW

File: agrona/src/main/java/org/agrona/io/DirectBufferDataInput.java [hunk 1]
```diff
@@ -0,0 +1,437 @@
...
+     */
+    public void wrap(final DirectBuffer buffer, final int offset, final int length)
+    {
+        if (null == buffer)
+        {
+            throw new NullPointerException("buffer cannot be null");
+        }
+
+        this.buffer = buffer;
+        this.length = length + offset;
```
Comment (mjpt777) [inline agrona/src/main/java/org/agrona/io/DirectBufferDataInput.java:92]: Validation of offset and length would be good to check against buffer.

**Comment by tkowalcz** [comment]
I have committed changes addressing review comments. 

Btw. I'm pretty sure that condition from `UnsafeBuffer::boundsCheckWrap`:

```java
 if ((offset > capacity - length) || (length > capacity - offset))
{
}
```
can be simplified to have just one of these inequalities as they are equivalent.

**Comment by mjpt777** [comment]
You should add validation of all the wrap parameters.

**Comment by tkowalcz** [comment]
You are right. Sorry. Pushed now (includes validation).

**Comment by tkowalcz** [comment]
I don't think that will work. I'll create second PR for this.

**Pull request merged**